skimage/transform/tests/test_geometric.py

```diff
@@ -544,6 +544,39 @@ def test_union_differing_types():
         tform1.__add__(tform2)
 
 
+@pytest.mark.parametrize(
+    "tform",
+    [
+        ProjectiveTransform(matrix=np.random.rand(3, 3)),
+        AffineTransform(scale=(0.1, 0.1), rotation=0.3),
+        EuclideanTransform(rotation=0.9, translation=(5, 5)),
+        SimilarityTransform(scale=0.1, rotation=0.9),
+        EssentialMatrixTransform(
+            rotation=np.eye(3), translation=(1 / np.sqrt(2), 1 / np.sqrt(2), 0)
+        ),
+        FundamentalMatrixTransform(
+            matrix=EssentialMatrixTransform(
+                rotation=np.eye(3), translation=(1 / np.sqrt(2), 1 / np.sqrt(2), 0)
+            ).params
+        ),
+        ((t := PiecewiseAffineTransform()).estimate(SRC, DST) and t),
```

Comment:
This is a slightly unobvious way to initialize the parameters of PiecewiseAffineTransform because the constructor does not accept arguments and `estimate` does not return the transform. This corresponds to:
```python
tform = PiecewiseAffineTransform()
tform.estimate(SRC, DST)
tform
```

Reply:
Huh? I thought I replied to this. I'm definitely either for adding a comment here or moving this bit into a helper function. It's a really cool and smart piece of code but probably not something that's easy to read for most people including me in a few months. :sweat_smile: 

I totally had to learn something new about how `and` works in Python! From https://docs.python.org/3/reference/expressions.html#boolean-operations

> Note that neither and nor or restrict the value and type they return to False and True, but rather return the last evaluated argument. This is sometimes useful, e.g., if s is a string that should be replaced by a default value if it is empty, the expression s or 'foo' yields the desired value. Because not has to create a new value, it returns a boolean value regardless of the type of its argument (for example, not 'foo' produces False rather than ''.)